eth/watchers/roundswatcher.go

```diff
@@ -0,0 +1,140 @@
+package watchers
+
+import (
+	"fmt"
+	"math/big"
+	"sync"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/eth/blockwatch"
+)
+
+// RoundsWatcher is a type alias for a thread safe in-memory cache that watches on-chain events for new initialized rounds
+type RoundsWatcher struct {
+	mu                       sync.RWMutex
+	lastInitializedRound     *big.Int
+	lastInitializedBlockHash [32]byte
+
+	quit chan struct{}
+
+	watcher BlockWatcher
+	lpEth   eth.LivepeerEthClient
+	dec     *EventDecoder
+}
+
+// NewRoundsWatcher creates a new instance of RoundsWatcher and sets the initial cache through an RPC call to an ethereum node
+func NewRoundsWatcher(roundsManagerAddr ethcommon.Address, watcher BlockWatcher, lpEth eth.LivepeerEthClient) (*RoundsWatcher, error) {
+	dec, err := NewEventDecoder(roundsManagerAddr, contracts.RoundsManagerABI)
+	if err != nil {
+		return nil, fmt.Errorf("error creating decoder: %v", err)
+	}
+
+	return &RoundsWatcher{
+		quit:    make(chan struct{}),
+		watcher: watcher,
+		lpEth:   lpEth,
+		dec:     dec,
+	}, nil
+}
+
+// LastInitializedRound gets the last initialized round from cache
+func (rw *RoundsWatcher) LastInitializedRound() *big.Int {
+	rw.mu.RLock()
+	defer rw.mu.RUnlock()
+	return rw.lastInitializedRound
+}
+
+// LastInitializedBlockHash returns the blockhash of the block the last round was initiated in
+func (rw *RoundsWatcher) LastInitializedBlockHash() [32]byte {
+	rw.mu.RLock()
+	defer rw.mu.RUnlock()
+	return rw.lastInitializedBlockHash
+}
+
+func (rw *RoundsWatcher) setLastInitializedRound(round *big.Int, hash [32]byte) {
+	rw.mu.Lock()
+	defer rw.mu.Unlock()
+	rw.lastInitializedRound = round
+	rw.lastInitializedBlockHash = hash
+}
+
+// Watch the blockwatch subscription for NewRound events
+func (rw *RoundsWatcher) Watch() {
+	lr, err := rw.lpEth.LastInitializedRound()
+	if err != nil {
```

Comment:
If any of the initial requests to populate the cache fail then we should probably propagate the error to the caller because there is a good chance that there is a problem with the remote ETH node and because otherwise if we continue there will be a period of time where any `RoundWatcher` user will get nil back for the last initialized round which can trigger a nil pointer error.

If we return an error from `Watch()` then we can handle any errors returned. But, similar to my comment about handling errors from the block watcher, I think we can address that in a separate PR.